### PR TITLE
controller: Refactor tests to use controller client.

### PR DESCRIPTION
**Pull request by archSeer**
Signed-off-by: Blaž Hrastnik <blaz.hrast@gmail.com>

### CODE REVIEW

File: controller/controller_test.go [hunk 1]
```diff
@@ -49,73 +49,14 @@ func (s *S) SetUpSuite(c *C) {
...
-		return res, fmt.Errorf("Unexpected status code %d", res.StatusCode)
-	}
-	return res, json.NewDecoder(res.Body).Decode(data)
-}
-
-func (s *S) Delete(path string) (*http.Response, error) {
-	req, err := http.NewRequest("DELETE", s.srv.URL+path, nil)
-	if err != nil {
-		return nil, err
+		c.Fatal(err)
```
Comment (titanous) [inline controller/controller_test.go]: `c.Assert(err, IsNil)`

File: controller/controller_test.go [hunk 2]
```diff
@@ -240,29 +165,24 @@ func (s *S) TestProtectedApp(c *C) {
...
-		procs  map[string]int
-		status int
+		procs map[string]int
+		check Checker
 	}{
-		{nil, 400},
-		{map[string]int{"web": 1}, 400},
-		{map[string]int{"worker": 1, "web": 0}, 400},
-		{map[string]int{"worker": 1, "web": 1}, 200},
+		{nil, Not(IsNil)},
```
Comment (titanous) [inline controller/controller_test.go]: `NotNil`

**Comment by archSeer** [comment]
Addressed and rebased on master.

**Comment by titanous** [comment]
LGTM pending tests passing.

**Pull request merged**